frontend/src/relay/auth.ts

```diff
@@ -0,0 +1,72 @@
+import { fetchQuery, graphql, GraphQLTaggedNode } from "react-relay";
+import { OperationType } from "relay-runtime";
+
+import { bug } from "../util/err";
+import CONFIG from "../config";
+import { environment } from ".";
+import { authLinkTokenQuery } from "./__generated__/authLinkTokenQuery.graphql";
+import { authUploadTokenQuery } from "./__generated__/authUploadTokenQuery.graphql";
+
+/**
+ * Authenticate an external link using a JWT.
+ * Returns a new `URL`.
+ */
+export const authenticateLink = async (link: URL | string): Promise<URL> => {
+    const query = graphql`
+        query authLinkTokenQuery {
+            externalLinkJwt
+        }
+    `;
+    const result = await getJwt<authLinkTokenQuery>(query);
```

Comment:
Shouldn't this query also only be done when `CONFIG.auth.preAuthExternalLinks` is true?

Reply:
Totally, good catch, I amended the commit! :D